src/librustc/ty/structural_impls.rs

```diff
@@ -839,14 +839,17 @@ impl<'tcx> TypeFoldable<'tcx> for ty::ExistentialProjection<'tcx> {
 
 impl<'tcx> TypeFoldable<'tcx> for ty::ProjectionTy<'tcx> {
     fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
+        let folded_trait_ref = self.trait_ref(folder.tcx()).fold_with(folder);
```

Comment:
Uhm I think I removed on master recently. So no, it wasn't needed, you can just fold substs.